### PR TITLE
chore: synth node should use a valid user agent

### PR DESCRIPTION
**Pull request by Rqnsom**
Three small items:
```
    chore: synth node should use a valid user agent

    The synth node should pretend to be a valid node instance,
    so, in this case, let's use MagicBean:5.4.2 and pretend to
    be a zcashd node.
```
```
    chore: change trace for receiving messages

    - trace log level for incoming messages should be `info`
      so it's easier to observe in the logs.
```
```
    chore: add a .gitignore for the synth node bin
```

### CODE REVIEW

File: src/protocol/payload/version.rs [hunk 1]
```diff
@@ -51,7 +51,8 @@ impl Version {
                 addr: addr_from,
             },
             nonce: Nonce::default(),
-            user_agent: VarStr(String::from("")),
+            // Let's pretend to be a ZCashd node 5.4.2
+            user_agent: VarStr(String::from("MagicBean:5.4.2")),
```
Comment (zeapoz) [inline src/protocol/payload/version.rs:55]: Let's make this a constant instead

**Review by kylegranger — APPROVED** [review]
LGTM 🚀

**Review by zeapoz — APPROVED** [review]
LGTM :+1:

File: src/protocol/payload/version.rs (hunk 1 above)
Comment (Rqnsom) [reply src/protocol/payload/version.rs]: Done

But usually, I follow a rule: if a magic value appears only once, creating a const is not necessary, it's just an overhead

**Pull request merged**